### DISCUSSION TITLE
Polar metrics - unexpected behavior in negative indices

### DISCUSSION THREAD

**Discussion by Antguz** [davidfrantz/force#237] [Q&A]
Dear FORCE community,

I am using polar metrics and the CCI to identify the decay of the physiological performance of trees on the land. The CCI is an index that ranges between -1 to 1; typical values are 0.4 at the peak of the growing season and negative (~ -0.2) at late fall. When extracting several polar metrics, I noticed some metrics did not make any sense.

For instance, take this CCI time series pixel from a forest cover (Fig 1). It could be expected that the Value of Late Minimum (VLM) to be negative (Fig 2), but I always got 0. Likewise, several pixels of the Value of the Peak Amplitude also appear to be 0. In all the cases, the Value of Base Level is higher or equal than 0, but never negative. 

![ts](https://user-images.githubusercontent.com/7254767/197311082-06a8ce4e-652b-41f7-a8c6-27fb2e95a3d9.png)
Fig 1. CCI time series. 2000.1 correspond to 2016-01-01.

![VLM](https://user-images.githubusercontent.com/7254767/197311085-fd20abe7-e4b6-4c18-9ed3-1c37e1cdadb5.png)
Fig 2. VLM of CCI.

I noticed that this pattern always appear in vegetation indices that tend to have negative values such as NDWI, NDVI or CCI, but not on those with positive range such as kNDVI, CIre, or values from spectral unmixing. 

If this is a bug, I consider that it occurs at the estimation of the polar coordinates. When a VI is negative (Fig 3), PCX (Fig 4) and PCY turns 0 (Fig 5).

 ![TSI](https://user-images.githubusercontent.com/7254767/197311102-4cd67a48-184b-4624-af71-8e119a5bd420.png)
Fig 3. Time series for the corresponding PCX and PCY.

![PCX](https://user-images.githubusercontent.com/7254767/197311091-7a56068c-f3c3-4d10-8b8b-3ff57ce81636.png)
Fig 4. PCX coordinate of CCI.

![PCY](https://user-images.githubusercontent.com/7254767/197311094-2917d2ca-2e1a-4778-b831-79ad8031141b.png)
Fig 5. PCY coordinate of CCI.

I appreciate all the help that you can provide on this. If you want to take a deeper look at the imagery, please use this small [tile](https://drive.google.com/drive/folders/1MWrSZTZVOouc2agl93Xy_TP_pJkf4Qik?usp=sharing) as an example.

Thanks in advance!

AG. (Sorry for the long post)

**Comment by davidfrantz** [comment]
Hi @Antguz,

yes, you are entirely correct. We also noticed this recently when using it on MNDWI. The computation of the polar metrics does not really allow its usage with indices that extend into the negative.

One of the reasons for that is the detection of the key dates via a cumulative sum of the values.

That said, please only use with positive indices. This is simply a limitation of that method.

I guess I will need to put a note somewhere.

Cheers,
David

**Reply by Antguz** [reply]
Hi @davidfrantz,

Thanks for your response. 

It is a bit disappointing (I will not lie) since most of my work is focused on an index with negative values, CCI. It confuses me at the same time since most of the papers that I see that use polar metrics are likely to face negative values, such as NDVI in winter. I understand the reasoning behind cumsum and that was my first thought. But because the magnitude of values here does not matter, do you not think that a prior scale will do the job?

CCI and most if not all Normalize Difference indices (VI) ranges between -1 to 1 (-10000 to 10000 FORCE scale) (by math). Applying an upscale (eg., VI+10000) prior the cumsum and then back scale to the resulting metrics (e.g., VPS-10000) may do the job? Not sure this will affect estimation of the integrals, but theoretically speaking those resulting metrics that involve rates, values (back-scale), or days should not be affected.

This is a great module and I consider that restring it to positive indices will limit the use of more than 20 VIs already presented in FORCE.

Regards,

AG.

**Reply by davidfrantz** [reply]
Well, what should I say? Not every method works in every case...

Regarding NDVI: this is actually not a real-life issue, since NDVI does not become negative for most land surfaces, even in winter. It might be negative for water or snow surfaces, but these should be filtered out anyway. The 0-truncation will effectively take care of this.

We also discussed scaling - or rather offsetting - with @danschef for a paper of his. This is basically possible and you will get a mathematically sound estimate. But, be aware, this will affect estimates as the shape of the cumulative curve and that of the polar coordinates will be altered. It also affects dates (see 0.2 start of season threshold below).

Please see this image generated by the sample R code below.

Cheers,
David

![image](https://user-images.githubusercontent.com/32633788/198231487-ebc48d6c-983e-405a-b0b6-c84eca31cb60.png)

```
layout(matrix(1:6, 3, 2))

d <- 1:365
# some phenological curve
n <- (sin(seq(-0.5*pi, 1.5*pi, length.out = 365)) + 1) / 2.5
# "start of season" at 20% cumulative
ss <- which.min(abs(cumsum(n)/max(cumsum(n)) - 0.2))


plot(d, n, main = "regular plot")

plot(d, cumsum(n)/max(cumsum(n)), main = "cumulative plot")
abline(v = d[ss])

r <- d/365*2*pi
x <- n*cos(r)
y <- n*sin(r)

plot(x, y, xlim = c(-1,1), ylim = c(-1,1), main = "polar plot")
abline(h=0)
abline(v=0)


n <- (sin(seq(-0.5*pi, 1.5*pi, length.out = 365)) + 1) / 2.5 + 1
ss <- which.min(abs(cumsum(n)/max(cumsum(n)) - 0.2))

plot(d, n, main = "regular plot, offsetted")

plot(d, cumsum(n)/max(cumsum(n)), main = "cumulative plot, offsetted")
abline(v = d[ss])

r <- d/365*2*pi
x <- n*cos(r)
y <- n*sin(r)

plot(x, y, xlim = c(-2,2), ylim = c(-2,2), main = "polar plot, offsetted")
abline(h=0)
abline(v=0)
```

**Reply by Antguz** [reply]
Thanks for this enriching discussion.

I was also playing with the Brooks et al. code and noticed that the offset is likely to change the detection of the days as you pointed out. That was my bad call. 

However, I also noticed that by doing normalization between 0 and 1, "`(x - max(x)) / (max(x) - min(x))`", the detection of days seems more reasonable than time series with negative values (at least for my samples). For instance, take a look at the third column from your example above; in this case, the days remain the same.

![normalized](https://user-images.githubusercontent.com/7254767/198335052-101a9757-14aa-42b0-b875-5ce5be2bd3c5.png)

```
n <- (sin(seq(-0.5*pi, 1.5*pi, length.out = 365)) + 1) / 2.5 + 1
n <- (n - min(n)) / (max(n) - min(n))
ss <- which.min(abs(cumsum(n)/max(cumsum(n)) - 0.2))
mins <- which.min(abs(cumsum(n)/max(cumsum(n)) - 0.5))
ls <- which.min(abs(cumsum(n)/max(cumsum(n)) - 0.8))

plot(d, n, main = "regular plot, normalized")

plot(d, cumsum(n)/max(cumsum(n)), main = "cumulative plot, normalized")
abline(v = d[ss]) #123 day
abline(v = d[mins]) #182 day
abline(v = d[ls]) #242 day

r <- d/365*2*pi
x <- n*cos(r)
y <- n*sin(r)

plot(x, y, xlim = c(-2,2), ylim = c(-2,2), main = "polar plot, normalized")
abline(h=0)
abline(v=0)
```

I guess this normalization will be more sensitive to covers with lower values like grasslands. In addition, I guess that the back transformation will require to carry two arguments.

Regards,

AG.
